### PR TITLE
Add the changes for Coq 8.16.0.

### PR DESCRIPTION
**Pull request by ppedrot**
As per the CEP this was generated by the corresponding script and put in a new section in the changes.

### CODE REVIEW

File: doc/sphinx/changes.rst [hunk 1]
```diff
@@ -8,6 +8,519 @@ Recent changes
...
+
+Notations
+^^^^^^^^^
+
+- **Removed:**
+  ``_`` in ``ident`` entries in notations, which was deprecated
+  in favor of ``name`` in 8.13. When you see messages like
+  ``Error: Notation "[ rel _ _ : _ | _ ]" is already defined at level 0
+  with arguments name, name, constr, constr while it is now required to be
+  at level 0 with arguments ident, ident, constr, constr.``,
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:113]: This long error message is shown entirely on a single line: https://coq.gitlab.io/-/coq/-/jobs/2501178726/artifacts/_install_ci/share/doc/coq/sphinx/html/changes.html#id28

We should find a way of making it break (or manually break it) on several lines.

File: doc/sphinx/changes.rst [hunk 2]
```diff
@@ -8,6 +8,519 @@ Recent changes
...
+  ``coqdep`` accepts ``-m META`` and use the file to resolve plugin names to
+  actual file names.
+  (`#15220 <https://github.com/coq/coq/pull/15220>`_,
+  fixes `#7698 <https://github.com/coq/coq/issues/7698>`_,
+  by Enrico Tassi).
+- **Changed:**
+  Minimum supported zarith version is now 1.11
+  (`#15483 <https://github.com/coq/coq/pull/15483>`_
+  and `#16005 <https://github.com/coq/coq/pull/16005>`_
+  and `#16030 <https://github.com/coq/coq/pull/16030>`,
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:494]: Broken link syntax:

```suggestion
  and `#16030 <https://github.com/coq/coq/pull/16030>`_,
```

File: doc/sphinx/changes.rst [hunk 3]
```diff
@@ -8,6 +8,519 @@ Recent changes
...
+- **Changed:**
+  Extraction from :g:`nat` to OCaml :g:`int` uses Stdlib instead of Pervasives
+  (`#15333 <https://github.com/coq/coq/pull/15333>`_,
+  by Rudy Nicolo Peterson).
+- **Changed:**
+  the ``signature`` scope of ``Classes.CMorphisms`` into ``signatureT``
+  (`#15446 <https://github.com/coq/coq/pull/15446>`_,
+  by Olivier Laurent).
+- **Changed:**
+  typeclass instances `Permutation_app'` and `Permutation_cons` are `#[export]` instead of `#[global]`
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:405]: At the very least:
```suggestion
  typeclass instances `Permutation_app'` and `Permutation_cons` are `#[` :attr:`export` `]` instead of `#[` :attr:`global` `]`
```

But I would even suggest rewording further into:
```suggestion
  the locality of typeclass instances `Permutation_app'` and `Permutation_cons` from :attr:`global` to :attr:`export`
```

**Review by Zimmi48 — COMMENTED** [review]
These are the few things that jump to my eyes. I will try to do a proper read through today.

The milestone was wrong since this PR needs to be backported (obviously).

File: doc/sphinx/changes.rst [hunk 4]
```diff
@@ -8,6 +8,519 @@ Recent changes
...
+
+.. contents::
+   :local:
+
+Kernel
+^^^^^^
+
+- **Changed:**
+  Fixpoints are now expected to be guarded even in subterms erasable
+  by reduction, thus getting rid from an artificial obstacle
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:25]:
```suggestion
  by reduction, thus getting rid of an artificial obstacle
```

File: doc/sphinx/changes.rst [hunk 5]
```diff
@@ -8,6 +8,519 @@ Recent changes
...
+  (`#14137 <https://github.com/coq/coq/pull/14137>`_,
+  fixes `#4632 <https://github.com/coq/coq/issues/4632>`_
+  and `#5384 <https://github.com/coq/coq/issues/5384>`_
+  and `#5521 <https://github.com/coq/coq/issues/5521>`_
+  and `#6278 <https://github.com/coq/coq/issues/6278>`_
+  and `#7675 <https://github.com/coq/coq/issues/7675>`_
+  and `#8739 <https://github.com/coq/coq/issues/8739>`_
+  and `#11011 <https://github.com/coq/coq/issues/11011>`_
+  and `#12240 <https://github.com/coq/coq/issues/12240>`_
+  and `#15279 <https://github.com/coq/coq/issues/15279>`_,
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:193]:
```suggestion
  fixes `#4632 <https://github.com/coq/coq/issues/4632>`_,
  `#5384 <https://github.com/coq/coq/issues/5384>`_,
  `#5521 <https://github.com/coq/coq/issues/5521>`_,
  `#6278 <https://github.com/coq/coq/issues/6278>`_,
  `#7675 <https://github.com/coq/coq/issues/7675>`_,
  `#8739 <https://github.com/coq/coq/issues/8739>`_,
  `#11011 <https://github.com/coq/coq/issues/11011>`_,
  `#12240 <https://github.com/coq/coq/issues/12240>`_,
  and `#15279 <https://github.com/coq/coq/issues/15279>`_,
```

File: doc/sphinx/changes.rst [hunk 6]
```diff
@@ -8,6 +8,519 @@ Recent changes
...
+Tactic language
+^^^^^^^^^^^^^^^
+
+- **Changed:**
+  Ltac `match` does not fail when the term to match contains an unfolded primitive projection
+  (`#15559 <https://github.com/coq/coq/pull/15559>`_,
+  fixes `#15554 <https://github.com/coq/coq/issues/15554>`_,
+  by Gaëtan Gilbert).
+- **Added:**
+  ``Ltac2`` understands :n:`toplevel_selector` and obeys :opt:`Default Goal Selector`.
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:229]:
```suggestion
  ``Ltac2`` understands :token:`toplevel_selector` and obeys :opt:`Default Goal Selector`.
```

File: doc/sphinx/changes.rst [hunk 7]
```diff
@@ -8,6 +8,519 @@ Recent changes
...
+  Ltac2 `apply` and `eapply` not unifying with implicit arguments;
+  unification inconsistent with `exact` and `eexact`
+  (`#15741 <https://github.com/coq/coq/pull/15741>`_,
+  by Ramkumar Ramachandra).
+
+SSReflect
+^^^^^^^^^
+
+- **Fixed:**
+  :tacn:`have`, :tacn:`suff` and :tacn:`wlog` support goals in `sProp`
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:247]:
```suggestion
  :tacn:`have`, :tacn:`suff` and :tacn:`wlog` support goals in `SProp`
```

File: doc/sphinx/changes.rst [hunk 8]
```diff
@@ -8,6 +8,519 @@ Recent changes
...
+  :cmd:`Abort` no longer takes an :n:`@ident` as an argument (it has been ignored since 8.5)
+  (`#15669 <https://github.com/coq/coq/pull/15669>`_,
+  by Gaëtan Gilbert).
+- **Removed:**
+  `Simplex` flag, that was deprecated in 8.14.
+  :tacn:`lia` and :tacn:`lra` will always use the simplex solver (that was already the default behaviour).
+  (`#15690 <https://github.com/coq/coq/pull/15690>`_,
+  by Frédéric Besson).
+- **Deprecated:**
+  :cmd:`Add LoadPath` and :cmd:`Add Rec LoadPath`
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:291]: We should probably repeat here the message that we have put elsewhere in the doc:

>If this command is an important feature for you, please open an issue at https://github.com/coq/coq/issues and explain your workflow

**Comment by Zimmi48** [comment]
@SkySkimmer has declared that he's planning an 8.15.2 release for this week / early next week. So we should delay this PR to merge first the one with the 8.15.2 changelog consolidation.

**Comment by Zimmi48** [comment]
Actually, given that there is only one PR in the list here that also belongs to 8.15.2, it would suffice to remove it from the PR here to allow merging it in any order (`doc/changelog/04-tactics/15866-inuit-solver.rst`):

```rst
- **Added:**
  :tacn:`intuition` and :tacn:`dintuition` use ``Tauto.intuition_solver`` (defined as ``auto with *``) instead of hardcoding ``auto with *``.
  This makes it possible to change the default solver with ``Ltac Tauto.intuition_solver ::= ...``
  (`#15866 <https://github.com/coq/coq/pull/15866>`_,
  fixes `#7725 <https://github.com/coq/coq/issues/7725>`_,
  by Gaëtan Gilbert).
```

File: doc/sphinx/changes.rst [hunk 9]
```diff
@@ -8,6 +8,515 @@ Recent changes
...
+
+Command-line tools
+^^^^^^^^^^^^^^^^^^
+
+- **Added:**
+  coq_makefile variable `COQPLUGININSTALL` to configure the installation of ML plugins
+  (`#15788 <https://github.com/coq/coq/pull/15788>`_,
+  by Cyril Cohen and Enrico Tassi).
+- **Added:**
+  Added -bytecode-compiler (yes|no) flag for ``coqchk`` enabling
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:365]: Could also be:
```suggestion
  Added :n:`-bytecode-compiler {| yes | no }` flag for ``coqchk`` enabling
```

File: doc/sphinx/changes.rst [hunk 10]
```diff
@@ -8,6 +8,515 @@ Recent changes
...
+  :tacn:`vm_compute` during checks, which is off by default.
+  (`#15886 <https://github.com/coq/coq/pull/15886>`_,
+  by Ali Caglayan).
+- **Fixed:**
+  :n:`coqdoc` confused by the presence of command :n:`Load` in a file
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:370]:
```suggestion
  ``coqdoc`` confused by the presence of command :cmd:`Load` in a file
```

File: doc/sphinx/changes.rst [hunk 11]
```diff
@@ -8,6 +8,515 @@ Recent changes
...
+  fixes `#15497 <https://github.com/coq/coq/issues/15497>`_,
+  by Hugo Herbelin).
+
+CoqIDE
+^^^^^^
+
+- **Changed:**
+  Bumped lablgtk3 lower bound to 3.1.2
+  (`#15947 <https://github.com/coq/coq/pull/15947>`_,
+  by Pierre-Marie Pédrot).
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:381]: Since this is the only item in CoqIDE, we could also move this to the "Infrastructure and dependencies" section.

File: doc/sphinx/changes.rst [hunk 12]
```diff
@@ -8,6 +8,515 @@ Recent changes
...
+Infrastructure and dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- **Changed:**
+  Load plugins using `findlib <http://projects.camlcity.org/projects/findlib.html>`_.
+  This requires projects built with ``coq_makefile`` to either provide a
+  hand written ``META`` file or use the ``-generate-meta-for-package`` option
+  when applicable. As a consequence :cmd:`Declare ML Module` now uses plugin
+  names according to ``findlib``, e.g. `coq-aac-tactics.plugin`.
+  ``coqdep`` accepts ``-m META`` and use the file to resolve plugin names to
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:481]:
```suggestion
  ``coqdep`` accepts ``-m META`` and uses the file to resolve plugin names to
```

File: doc/sphinx/changes.rst [hunk 13]
```diff
@@ -8,6 +8,515 @@ Recent changes
...
+- **Changed:**
+  Bump the minimum OCaml version to 4.09.0.
+  As a consequence the minimum supported ocamlfind version is now
+  1.8.1 (`#15947 <https://github.com/coq/coq/pull/15947>`_
+  and `#16046 <https://github.com/coq/coq/pull/16046>`_,
+  fixes `#14260 <https://github.com/coq/coq/issues/14260>`_
+  and `#16015 <https://github.com/coq/coq/pull/16015>`_,
+  by Pierre-Marie Pédrot and Théo Zimmermann).
+
+Miscellaneous
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:502]: As frequently happens, this section is all about extraction, so I suggest renaming it (and moving the items from the Standard library section that are relevant to extraction here).

File: doc/sphinx/changes.rst [hunk 14]
```diff
@@ -8,6 +8,515 @@ Recent changes
...
+- **Changed:**
+  `ExtrOCamlInt63` no longer extracts `comparison` to `int` in OCaml;
+  the extraction of `Uint63.compare` and `Sint63.compare` was also adapted accordingly
+  (`#15294 <https://github.com/coq/coq/pull/15294>`_,
+  fixes `#15280 <https://github.com/coq/coq/issues/15280>`_,
+  by Li-yao Xia).
+- **Changed:**
+  Extraction from :g:`nat` to OCaml :g:`int` uses Stdlib instead of Pervasives
+  (`#15333 <https://github.com/coq/coq/pull/15333>`_,
+  by Rudy Nicolo Peterson).
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:395]: I.e., these two.

File: doc/sphinx/changes.rst [hunk 15]
```diff
@@ -8,6 +8,515 @@ Recent changes
...
+  by Olivier Laurent).
+- **Deprecated:**
+  `P_Rmin`: use more general `Rmin_case` instead
+  (`#15388 <https://github.com/coq/coq/pull/15388>`_,
+  fixes `#15382 <https://github.com/coq/coq/issues/15382>`_,
+  by Olivier Laurent).
+- **Added:**
+  More extraction definitions for division and comparison of Z and N
+  (`#15098 <https://github.com/coq/coq/pull/15098>`_,
+  by Li-yao Xia).
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:438]: And this one.

File: doc/sphinx/changes.rst (hunk 1 above)
Comment (ppedrot) [reply doc/sphinx/changes.rst:113]: No idea about how to do that.

Comment (ppedrot) [reply doc/sphinx/changes.rst]: I pushed a commit to try that. Let's see.

Comment (ppedrot) [reply doc/sphinx/changes.rst]: Seems to have worked: https://coq.gitlab.io/-/coq/-/jobs/2522451683/artifacts/_install_ci/share/doc/coq/sphinx/html/changes.html#id30

**Review by Zimmi48 — APPROVED** [review]
@coqbot merge now
